### ISSUE TITLE
Bug: [comma-spacing] conflicts with array-bracket-spacing

### ISSUE THREAD

**Issue by alecrobertson1** [typescript-eslint/typescript-eslint#6874]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.0.2&sourceType=module&code=NocgbghgNgrgpiANAAnNeAmJyC6Q&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQIbWlwE8BaAI0IGMBrRAFxOQAddKBLAOwHN0pECA9tEgAacNkgABOkSYpK0NkwYp4nOgHpKAgLY7cjFu269I-aEMjiAviGtA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
['value', 'value2',]
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "array-bracket-spacing": "error",
    "@typescript-eslint/comma-spacing": "error"
  }
};
```

### Expected Result

[comma-spacing] should not throw an error, as there should not be a space between a trailing comma in an array and the closing "]" bracket.

### Actual Result

[comma-spacing] throws an error, "A space is required after ','".

### Additional Info

This issue was fixed in ESLint's [comma-spacing] rule here: https://github.com/eslint/eslint/pull/16113.

**Comment by Josh-Cena** [comment]
Yes, I think we should sync with upstream
